runtime/src/bank.rs

```diff
@@ -442,6 +442,7 @@ impl Bank {
             new.ancestors.insert(p.slot(), i + 1);
         });
 
+        new.update_slot_hashes(parent.slot(), parent.hash());
```

Comment:
Strictly speaking, this parent bank's book keeping is duplicated across all the child (= forked) banks; but I think this is fine to avoid complicated mutable bank hash management otherwise.